miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt

```diff
@@ -161,17 +162,40 @@ abstract class MiniAppMessageBridge {
                 object : TypeToken<Permission>() {}.type
             )
 
-            requestPermission(
-                MiniAppPermissionType.getValue(permissionParam.permission)
-            ) { isGranted -> onRequestPermissionsResult(
-                callbackId = callbackObj.id,
-                isGranted = isGranted
-            ) }
+            val permissionName = MiniAppPermissionType.getValue(permissionParam.permission)
+
+            if (permissionName.type == MiniAppPermissionType.LOCATION.type) {
```

Comment:
@corycaywood I have added a solution for spec: `when a user tries to request location, we should first check that the "rakuten.miniapp.device.location" permission has been granted.`.  Do you think, we need to check it inside SDK? Or, I should move this implementation to HostApp?

Because, I was also reading a note in Jira: `This permission should be handled separately from the native Location permission which we handle via requestPermission`

p/s: I'll add unit test after confirming this fact.

Reply:
@khairul-alam-licon Sorry, this wasn't very clear. I don't think we need to modify any of the current behavior of `onRequestPermission`. From the native SDK side, we should treat the device location permission request and the custom location permission request as completely separate and unrelated. The Host App also shouldn't have to worry about this.

I think we can add some functionality to the JS SDK to handle the interaction between the two. So I think the requirement for `when a user tries to request location, we should first check that the "rakuten.miniapp.device.location" permission has been granted.` can be handled in the JS SDK instead.

Reply:
Okay, I am updating this part.